test/integration/autonomous_upstream.cc

```diff
@@ -137,6 +143,12 @@ void AutonomousUpstream::setResponseHeaders(
   response_headers_ = std::move(response_headers);
 }
 
+void AutonomousUpstream::setResponseMetadata(
```

Comment:
Technically metadata can come anywhere in the stream, right? Maybe call this `setPreResponseHeadersMetadata` or something like that in case in the future we want to try different interleavings?

Reply:
done